aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -132,18 +129,17 @@ bool is_nonzero(const Tensor& self) {
   } else if (localScalar.isBoolean()) {
     return localScalar.to<bool>();
   }
-  AT_ERROR("expected non-Tensor backend scalar");
+  TORCH_INTERNAL_ASSERT(false, "Expected non-Tensor backend scalar");
```

Comment:
This should probably be TORCH_CHECK(false, ...

Reply:
The branches above have enumerated all possibilities (note that there is no return value specified down here). My understanding is that this line serves as a safeguard in case there is a new scalar type added, a test error would emerge here. Hence this check is for devs, not users, and hence `TORCH_INTERNAL_ASSERT`.